nucypher/network/server.py

```diff
@@ -222,13 +225,40 @@ def set_policy(self, id_as_hex, request: Request):
         # TODO: Sign the arrangement here.  #495
         return  # TODO: Return A 200, with whatever policy metadata.
 
+    def revoke_arrangement(self, id_as_hex, request: Request):
+        """
+        REST endpoint for revoking/deleting a KFrag from a node.
+        TODO: Have Ursula return a signed receipt.
+        """
+        from nucypher.policy.models import RevocationNotice
+
+        revocation_notice = RevocationNotice.from_bytes(request.body)
+
+        try:
+            with ThreadedSession(self.db_engine) as session:
+                # Verify the Notice was signed by Alice
+                policy_arrangement = self.datastore.get_policy_arrangement(
```

Comment:
I think, better to verify before getting this from the DB.
Checking signature is slow, getting the arrangement from the DB is faster, so an attacker (who is not Alice) can figure out, who has a particular kfrag and who doesn't

Reply:
Ahh, we must get Alice's public key from the policy arrangement...
A different way would be to get it from the signature and compare.

Anyway, my point was about constant-timeness of this :-)

Reply:
I mean, sure, we can check an arbitrary signature next to it to give it some semblance of time parity.

But I think (and I don't want to speak for @tuxxy, but I think he somewhat agrees here based on our conversation last night) that we might be wandering into something like security theatre here.

What's the attack here?  A timing attack across a huge swath of the network to determine whether Ursula has an `Arrangement` matching a particular ID?

I'm not sure this is even particularly sensitive information (though I don't advocate shouting it from the mountaintops).

If an attacker has stolen a `TreasureMap` (and I'm not sure where else she'd glean the arrangement ID), then she already knows which Ursulas are in the `Policy`.

But yes, fine, let's check a blank sig in the negative case and call it a night.

Reply:
actually, _right now_ it's not sensitive at all because we do shout from the mountaintops when writing onto Ethereum blockchain...

Reply:
Rather than optimize for this (it's kinda like security theater), we should be using exponential backoff on the API endpoints anyway.